marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java

```diff
@@ -216,59 +220,74 @@ private void updateLatestChangeToProductsFromGithubRepo() {
-        if (FileType.META == file.getType()) {
-          transferComputedDataFromDB(product);
-          modifyProductByMetaContent(file, product);
+    groupGitHubFiles.forEach((key, value) -> {
+      for (var file : value) {
+        if (file.getStatus() == MODIFIED || file.getStatus() == ADDED) {
+          modifyProductMetaOrLogo(file, key);
         } else {
-          modifyProductLogo(ghFileEntity.getKey(), file, product, fileContent);
+          removeProductAndImage(file);
         }
       }
     });
   }
 
-  private void modifyProductLogo(String parentPath, GitHubFile file, Product product, GHContent fileContent) {
-    Product result;
-    switch (file.getStatus()) {
-      case MODIFIED, ADDED:
-        var searchCriteria = new ProductSearchCriteria();
-        searchCriteria.setKeyword(parentPath);
-        searchCriteria.setFields(List.of(MARKET_DIRECTORY));
-        result = productRepository.findByCriteria(searchCriteria);
-        if (result != null) {
-          Optional.ofNullable(imageService.mappingImageFromGHContent(result, fileContent, true)).ifPresent(image -> {
-            imageRepository.deleteById(result.getLogoId());
-            result.setLogoId(image.getId());
-            productRepository.save(result);
-          });
-        }
-        break;
-      case REMOVED:
-        result = productRepository.findByLogoId(product.getLogoId());
-        if (result != null) {
-          imageRepository.deleteAllByProductId(result.getId());
-          productRepository.deleteById(result.getId());
+  private void removeProductAndImage(GitHubFile file) {
+    if (FileType.META == file.getType()) {
+      String[] splitMetaJsonPath = file.getFileName().split("/");
+      String extractMarketDirectory = file.getFileName().replace(splitMetaJsonPath[splitMetaJsonPath.length - 1], "");
+      List<Product> productList = productRepository.findByMarketDirectory(extractMarketDirectory);
+      if (ObjectUtils.isNotEmpty(productList)) {
+        String productId = productList.get(0).getId();
+        productRepository.deleteById(productId);
+        imageRepository.deleteAllByProductId(productId);
+      }
+    } else {
+      List<Image> images = imageRepository.findByImageUrlEndsWithIgnoreCase(file.getFileName());
+      if (ObjectUtils.isNotEmpty(images)) {
+        Image currentImage = images.get(0);
+        productRepository.deleteById(currentImage.getProductId());
+        imageRepository.deleteAllByProductId(currentImage.getProductId());
+      }
+    }
+  }
+
+  private void modifyProductMetaOrLogo(GitHubFile file, String key){
+    Product product = new Product();
+    GHContent fileContent;
+    try {
+      fileContent = gitHubService.getGHContent(axonIvyMarketRepoService.getRepository(), file.getFileName(),
+          marketRepoBranch);
+    } catch (IOException e) {
+      log.error("Get GHContent failed: ", e);
+      return;
+    }
+    ProductFactory.mappingByGHContent(product, fileContent);
```

Comment:
Could you create a new method for the block code 269->275 and move it into the try statement?